File: dcp-solutions/2025-01-01/solution.php
```php
class Solution {
    public function reverseWords($s) {
        $words = explode(' ', $s);
        $reversed_words = array_reverse($words);
        $result = implode(' ', $reversed_words);
        return $result;
    }
}
```

Note that this PHP solution uses the `array_reverse` function to reverse the order of the words and then uses `implode` to join them back together into a single string.

File: dcp-solutions/2025-01-01/solution.ex
**Elixir Solution**
```markdown
defmodule StringReverser do
  @doc """
  Reverses words in a string.
  
  Args:
    s (String): The input string.
  
  Returns:
    String: The reversed string.
  """
  def reverse_words(s) do
    # Split the string into words
    words = String.split(s, " ")
    
    # Reverse each word and join them back together with spaces in between
    Enum.map(words, &reverse/1) |> Enum.join(" ")
  end

  @doc """
  Reverses words in a mutable string representation in-place.
  
  Args:
    s (String): The input string.
  """
  def reverse_in_place(s) do
    # Split the string into words
    words = String.split(s, " ")
    
    # Reverse each word and join them back together with spaces in between
    Enum.map(words, &reverse/1) |> Enum.join(" ") |> String.replace(~r/\s+/, " ", {:no_replace})
  end

private
defp reverse(word), do: String.reverse(word)
end
```